src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -425,6 +428,41 @@ public void testFileNameNoExtension() throws Exception {
                 getPath("InputImportControlFileNameNoExtension"), expected);
     }
 
+    @Test
+    public void testPathWithNonAsciiCharacters() throws Exception {
+        final Path dirWithNonAsciiCharacters = temporaryFolder.toPath()
+                .resolve("\\u00e6\\u00f8\\u00e5")
+                .resolve("com/puppycrawl/tools/checkstyle/checks/imports/importcontrol");
+        dirWithNonAsciiCharacters.toFile().mkdirs();
+        final Path configInDirWithNonAsciiCharacters = dirWithNonAsciiCharacters.resolve(
+                "InputImportControl8.java");
+        for (String file : List.of("InputImportControl8.java", "InputImportControlBroken.xml")) {
+            Files.copy(Paths.get(getPath(file)), dirWithNonAsciiCharacters.resolve(file));
+        }
+
+        try {
+            final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+            verifyWithInlineConfigParser(configInDirWithNonAsciiCharacters.toString(), expected);
+            assertWithMessage("Test should fail if exception was not thrown").fail();
```

Comment:
We need test that show that validation works.
https://github.com/checkstyle/checkstyle/blob/820dc2677136021c00413e688b26a7d7d01f06b2/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/importcontrol/InputImportControl8.java#L3

Can you put here full path after copy, it should be loaded.
Test should sounds easy: we cop file to folder with Unicode, we run  validation, all works. ( Same as as your issue sounds)